src/backend/distributed/commands/common.c

```diff
@@ -283,3 +285,53 @@ PreprocessDropDistributedObjectStmt(Node *node, const char *queryString,
 
 	return NodeDDLTaskList(NON_COORDINATOR_NODES, commands);
 }
+
+
+/*
+ * DropTextSearchDictObjectAddress returns list of object addresses in
+ * the drop tsdict statement.
+ */
+List *
+DropTextSearchDictObjectAddress(Node *node, bool missing_ok)
+{
+	DropStmt *stmt = castNode(DropStmt, node);
+
+	List *objectAddresses = NIL;
+
+	List *objName = NIL;
+	foreach_ptr(objName, stmt->objects)
+	{
```

Comment:
We will use address method (if the dist object has any) to check if we should run preprocess and postprocess steps. We will call the address method with a missing_ok flag set to true, so it will be always safe while running pg tests. In all cases missing_ok flag will be respected.

Reply:
We have not call any address method in postprocess so far if the statement is related to drop. As you said, we need to tighten the code to not allow to call address in postprocess for drop statements. I guess we can discuss that in 'alter domain drop constraint' PR that I will open soon.